File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        var visit = new HashSet<char>();
        int maxLen = 0;
        int l=0;
        int r=l;
        while(r<s.Count()){
            if(!visit.Contains(s[r])){
                visit.Add(s[r]);
                maxLen = Math.Max(maxLen, r-l+1);
                r++;
            }
            else{
                visit.Remove(s[l]);
                l++;
            }
        }
        return maxLen;
    }
}

File: Data Structures & Algorithms/house-robber/submission-2.java
class Solution {
    public int rob(int[] nums) {
        int a = 0, b = 0;
        for(int i=0;i<nums.length;i++){
            int c = Math.max(a+nums[i], b);
            a = b;
            b = c;
        }
        return Math.max(a,b);
    }
}
